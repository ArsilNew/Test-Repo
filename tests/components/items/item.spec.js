import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import { fireEvent, render } from "@testing-library/react";

import Item from "../../../src/components/items/item";

const item = { id: 1, label: "Hi" };

describe("Item", () => {
  test("default snapshot", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Item />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with valid item", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Item item={item} />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with item without id", () => {
    const itemWithoutId = { label: "Hi" };

    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Item item={itemWithoutId} />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with checked status", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Item checked={true} />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with indeterminate status", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Item indeterminate={true} />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with indeterminate and checked status", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Item indeterminate={true} checked={true} />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with border", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Item withBorder={true} />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with disabled", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Item disabled={true} />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with height", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Item height={20} />);
    expect(tree).toMatchSnapshot();
  });

  test("click will trigger onClick", () => {
    const onClick = jest.fn();
    const { container } = render(<Item onClick={onClick} />);
    fireEvent.click(container.querySelector("div.item"));
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  test("snapshot with group item", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Item item={item} group />);
    expect(tree).toMatchSnapshot();
  });
});
