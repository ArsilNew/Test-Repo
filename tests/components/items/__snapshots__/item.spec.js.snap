// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item default snapshot 1`] = `
<div
  className="item"
  onClick={undefined}
  style={
    Object {
      "height": 40,
    }
  }
>
  <WithStyles(Checkbox)
    checked={false}
    color="primary"
    disabled={false}
    indeterminate={false}
    type="checkbox"
  />
  <ItemLabel
    label=""
  />
</div>
`;

exports[`Item snapshot with border 1`] = `
<div
  className="item with_border"
  onClick={undefined}
  style={
    Object {
      "height": 40,
    }
  }
>
  <WithStyles(Checkbox)
    checked={false}
    color="primary"
    disabled={false}
    indeterminate={false}
    type="checkbox"
  />
  <ItemLabel
    label=""
  />
</div>
`;

exports[`Item snapshot with checked status 1`] = `
<div
  className="item selected"
  onClick={undefined}
  style={
    Object {
      "height": 40,
    }
  }
>
  <WithStyles(Checkbox)
    checked={true}
    color="primary"
    disabled={false}
    indeterminate={false}
    type="checkbox"
  />
  <ItemLabel
    label=""
  />
</div>
`;

exports[`Item snapshot with disabled 1`] = `
<div
  className="item disabled"
  onClick={undefined}
  style={
    Object {
      "height": 40,
    }
  }
>
  <WithStyles(Checkbox)
    checked={false}
    color="primary"
    disabled={true}
    indeterminate={false}
    type="checkbox"
  />
  <ItemLabel
    label=""
  />
</div>
`;

exports[`Item snapshot with group item 1`] = `
<div
  className="item with_grouping"
  onClick={undefined}
  style={
    Object {
      "height": 40,
    }
  }
>
  <ItemLabel
    label="Hi"
  />
</div>
`;

exports[`Item snapshot with height 1`] = `
<div
  className="item"
  onClick={undefined}
  style={
    Object {
      "height": 20,
    }
  }
>
  <WithStyles(Checkbox)
    checked={false}
    color="primary"
    disabled={false}
    indeterminate={false}
    type="checkbox"
  />
  <ItemLabel
    label=""
  />
</div>
`;

exports[`Item snapshot with indeterminate and checked status 1`] = `
<div
  className="item selected"
  onClick={undefined}
  style={
    Object {
      "height": 40,
    }
  }
>
  <WithStyles(Checkbox)
    checked={true}
    color="primary"
    disabled={false}
    indeterminate={true}
    type="checkbox"
  />
  <ItemLabel
    label=""
  />
</div>
`;

exports[`Item snapshot with indeterminate status 1`] = `
<div
  className="item"
  onClick={undefined}
  style={
    Object {
      "height": 40,
    }
  }
>
  <WithStyles(Checkbox)
    checked={false}
    color="primary"
    disabled={false}
    indeterminate={true}
    type="checkbox"
  />
  <ItemLabel
    label=""
  />
</div>
`;

exports[`Item snapshot with valid item 1`] = `
<div
  className="item"
  onClick={undefined}
  style={
    Object {
      "height": 40,
    }
  }
>
  <WithStyles(Checkbox)
    checked={false}
    color="primary"
    disabled={false}
    indeterminate={false}
    type="checkbox"
  />
  <ItemLabel
    label="Hi"
  />
</div>
`;
